src/errors.ts

```diff
@@ -0,0 +1,32 @@
+export enum AptosWalletErrorCode {
+  Unauthorized = 4100,
+  InternalError = -30001
+}
+
+export const AptosWalletErrors = Object.freeze({
+  [AptosWalletErrorCode.Unauthorized]: {
+    title: 'Unauthorized',
+    message: 'The requested method and/or account has not been authorized by the user.'
+  },
+  [AptosWalletErrorCode.InternalError]: {
+    title: 'Internal error',
+    message: 'Something went wrong within the wallet.'
+  }
+})
+
+export class AptosWalletError extends Error {
+  readonly code: number
+  readonly title: string
```

Comment:
Thoughts on calling this `status`?
Similarly to HTTP status and status code
400 Bad request
500 Internal server error
...